Thelounge - sqlite3 unable to load

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

The Lounge LXC

### 📂 What was the exact command used to execute the script?

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/the-lounge.sh)"

### 📝 Provide a clear and concise description of the issue.

I'm migrating from a vps to a lxc.
While migrating the data, I found out no logs were imported and the `/search` function wasn't enabled.

After discussing with a maintainer of the lounge on irc, it happens when sqlite3 isn't present/supported by the install.

Doing some diging in the service logs and it seems the sqlite3 bundle with the install doesn't install/load correctly.

I tried to install sqlite3 with npm, yarn and also recompile the package (from local source, from online sources), alas to no avail.

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 🔄 Steps to reproduce the issue.

Step1. Install the lxc with the given command on https://community-scripts.github.io/ProxmoxVE/scripts?id=the-lounge
Step2. Go into lxc shell
Step3. Type `service thelounge status`
Step4. Observe that the service is unable to load sqlite3 module

### ❌ Paste the full error output (if available).

```
Mar 27 20:07:30 the-lounge systemd[1]: Started thelounge.service - The Lounge (IRC client).
Mar 27 20:07:32 the-lounge thelounge[158]: 2025-03-27 19:07:32 [ERROR] Unable to load sqlite3 module. See https://github.com/mapbox/node-sq>
Mar 27 20:07:34 the-lounge thelounge[158]: 2025-03-27 19:07:34 [INFO] The Lounge v4.4.3 (Node.js 22.14.0 on linux x64)
Mar 27 20:07:34 the-lounge thelounge[158]: 2025-03-27 19:07:34 [INFO] Configuration file: /etc/thelounge/config.js
Mar 27 20:07:34 the-lounge thelounge[158]: 2025-03-27 19:07:34 [INFO] Available at http://[::]:9000/ in private mode
```

### 🖼️ Additional context (optional).

Don't need to tweak the config of thelounge to use sqlite3 to observe the issue, a fresh install shows it in the service logs.

--- a/ct/the-lounge.sh
+++ b/ct/the-lounge.sh
@@ -27,6 +27,13 @@ function update_script() {
     msg_error "No ${APP} Installation Found!"
     exit
   fi
+  if ! dpkg -l build-essential >/dev/null 2>&1; then
+    $STD apt-get update
+    $STD apt-get install -y build-essential
+  fi
+  if ! npm list -g node-gyp >/dev/null 2>&1; then
+    $STD npm install -g node-gyp
+  fi
   RELEASE=$(curl -s https://api.github.com/repos/thelounge/thelounge-deb/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
   if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
     msg_info "Stopping Service"

--- a/install/the-lounge-install.sh
+++ b/install/the-lounge-install.sh
@@ -14,7 +14,9 @@ network_check
 update_os
 
 msg_info "Installing Dependencies"
-$STD apt-get install -y gpg
+$STD apt-get install -y \
+  gpg \
+  build-essential
 msg_ok "Installed Dependencies"
 
 msg_info "Setting up Node.js Repository"
@@ -27,6 +29,7 @@ msg_info "Installing Node.js"
 $STD apt-get update
 $STD apt-get install -y nodejs
 $STD npm install --global yarn
+$STD npm install --global node-gyp
 msg_ok "Installed Node.js"
 
 msg_info "Installing The Lounge"

